crates/ecs/src/systems/mod.rs

```diff
@@ -710,14 +738,67 @@ impl<'a, Component: Debug + Send + Sync + 'static + Sized> SystemParameter
     }
 }
 
-/*
+/// Segment for [`SystemParameter`] without data.
+#[derive(Debug, Default)]
+pub struct UnitSegment<T: Default> {
+    item: PhantomData<T>,
+}
+
+impl<T: Default> UnitSegment<T> {
+    pub(crate) fn create_segments(segment_config: SegmentConfig) -> Vec<Self> {
+        match segment_config {
+            SegmentConfig::Single => {
+                vec![UnitSegment::default()]
+            }
+            SegmentConfig::Size { segment_count, .. } => {
+                vec![UnitSegment::default(); segment_count]
+            }
+        }
+    }
+}
+
+impl<T: Default> Clone for UnitSegment<T> {
+    fn clone(&self) -> Self {
+        Self { item: PhantomData }
+    }
+}
+
+impl<T: Default> IntoIterator for UnitSegment<T> {
+    type Item = T;
+    type IntoIter = impl Iterator<Item = Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        iter::from_fn(|| Some(T::default()))
+    }
+}
+
+/// Segment for [`Entity`]
+#[derive(Debug)]
+pub struct EntitySegment<'components> {
+    slices: Vec<&'components [Entity]>,
+}
+
+impl<'components> Clone for EntitySegment<'components> {
+    fn clone(&self) -> Self {
+        Self {
+            slices: self.slices.clone(),
+        }
+    }
+}
```

Comment:
How come this is necessary, instead of deriving it? Since both `Vec<T: Clone>: Clone` and `&T: Clone`, I'd imagine that it's possible to derive, but I might be missing something

Reply:
Same goes for `UnitSegment`

Reply:
The code generated with derive usually has unnecessary strict trait bounds when working with slices and `PhantomData`. This is why `ReadSegment` cannot derive `Clone`, filters cannot derive `Default` and `UnitSegment` cannot derive `Clone`. `EntitySegment` is the exception because `Entity` implements `Clone`, so in this case it would work.